packages/transaction-controller/src/TransactionController.ts

```diff
@@ -610,9 +619,94 @@ export class TransactionController extends BaseControllerV1<
       this.onBootCleanup();
     });
 
+    this.onNetworkStateChange = onNetworkStateChange;
+    this.isMultichainEnabled = isMultichainEnabled;
+    this.incomingTransactions = incomingTransactions;
+    this.getNetworkClientRegistry = getNetworkClientRegistry;
+
     this.onBootCleanup();
   }
 
+  /**
+   * Initializes `registry`, `nonceTracker`, `#multichainTrackingHelper`,
+   * `incomingTransactionHelper`, `pendingTransactionTracker`, and passes a
+   * callback to `onNetworkStateChange`.
+   *
+   * This relies on the provider and block tracker being defined which may not
+   * have been the case at the time the Transaction Controller was instantiated.
+   *
+   * @param passedProvider - Reference to the provider proxy object
+   * @param passedBlockTracker - Reference to the block tracker proxy object
+   */
+  delayedInit(passedProvider: Provider, passedBlockTracker: BlockTracker) {
```

Comment:
My main concern is that these helpers are fundamental to the operation of this controller as they are required not only to monitor transactions but to even add them.

Meaning an instance without them feels very limited in utility and we now have to be conscious of these two alternate states of initialisation which adds additional complexity to an already complex controller.

If the client doesn't have the necessary dependencies to initialise the controller, is that not by definition a client problem and so would a cleaner and simpler solution be to delay the initialisation of the controller on the client side until all the dependencies are available? This would also have the benefit of making the flows easier to predict and handle as the client would be able to create suitable errors if the controller isn't yet initialised but an action is triggered that requires it. 

Ultimately I worry this is over-coupling the client to the controller as the intent of the controllers to begin with is to encapsulate state and logic out of the clients, using the constructors to define the explicit dependencies they require to function.

Reply:
I agree. We should do our best to not place things in an invalid state on purpose, and this would include initializing a controller but leaving its provider, block tracker, or EthQuery instance unset until some future time. Delaying the initialization of the controller until we can pass a provider seems like a better approach. There's probably some context I'm missing here, so I'll see if I can study this further this week.